scripts/local-generate-svg.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+export SOURCECRED_DIRECTORY=$(pwd)/sourcecred_data
+SOURCECRED_GITHUB_TOKEN=`cat secrets/token`
+REPOS=`cat repositories.txt`
+
+cd mkweights
+yarn
+cd ..
+
+cat ./weights.toml | node mkweights > .weights.json
+
+cd sourcecred
+yarn install
+yarn backend
+for repo in $REPOS; do
+	SOURCECRED_GITHUB_TOKEN=$SOURCECRED_GITHUB_TOKEN node bin/sourcecred.js load $repo --weights ../.weights.json
```

Comment:
Could you walk me through what is going on here?
 
 - node mkweights create some original file, (and weights.toml is the configuration of weights?
 - yarn install installs, what does backend do?
 - then we iterate through the repos, and we issue a load command. Are the repos already included in weight.json (and we are loading) or are we adding data to weights.json for the repo?

Probably some comments in these sections would fit the bill, so others / future you can come back and remember what it does :)

Reply:
> node mkweights create some original file, (and weights.toml is the configuration of weights?

Yes. Sourcecred has a json weights format that is more verbose than the toml file format I'm using. I created the toml format as a more intuitive one to discuss. And lines 7-11 are to convert the toml format to the json format.

> yarn install installs, what does backend do?

Yarn has a cli syntax `yarn <script_name>` so this is the equivalent of `npm run <script_name>`. In this case the script `backend` is [defined here](https://github.com/Beanow/sourcecred/blob/d98abe0c235df8502f8001f19feb2e8b93244309/package.json#L90). Which will use webpack and babel to transliterate the sourcecred source code (using flow and esm) to javascript code that can be executed by node 10 or 12.

> then we iterate through the repos, and we issue a load command. Are the repos already included in weight.json (and we are loading) or are we adding data to weights.json for the repo?

The load command fetches github data and runs the sourcecred scoring algorithm. We pass the weights arguments as configuration values. These weights essentially quantify: "How important is a comment? How important is a commit? How important is a PR?" ... etc.

---

That is the context to what these commands entail.

In broad stroke the workflow is:
- Convert our simplified weights.toml to the weights.json format used by SourceCred.
- yarn install + yarn backend to set up SourceCred.
- `sourcecred/bin/sourcecred.js load ...` to fetch github data and score it using our weights.
- `sourcecred/bin/sourcecred.js scores ...` to export those scores in a format we can consume.
- `widgets/bin/contributor-wall-svg.js` to consume those scores and generate the SVG.

Reply:
One final question here - just want to check that there is no kind of PI in the GitHub data?

Reply:
That is a good question. My assumption would be that because no special permissions are required to fetch user info it could only be public data. However many people people use their full name and an email and make it public. So that may be PI either way. I don't think they are stored, only the ID and username for graphs and avatars and username for widgets. Though I would need to check.